.github/workflows/ci-run-on-main.yml

```diff
@@ -0,0 +1,76 @@
+name: Push to Main CI
```

Comment:
I would rename this file to something more descriptive and has TLM in the name: `ci-tlm-main.yml` ?

Reply:
I wanted to keep it general in case we add other non-tlm related tests here but I have no strong opinions